torch/testing/_internal/common_optimizers.py

```diff
@@ -99,17 +99,24 @@ def __init__(
         self.error_regex = error_regex
 
 
+# Typing for every optim_inputs_func
+class OptimInputsFunc(Protocol):
```

Comment:
Why does this need to be a protocol? JIT support?

Reply:
It probably doesn't! I just thought that in this typing structure you're supposed to use a Protocol, but I might've learned incorrectly and have been unnecessarily adding Protocol

Reply:
Just got rid of it, thanks for pointing that out!